src/ripple/protocol/impl/STParsedJSON.cpp

```diff
@@ -140,14 +141,14 @@ static Json::Value singleton_expected (std::string const& object,
 
 // This function is used by parseObject to parse any JSON type that doesn't
 // recurse.  Everything represented here is a leaf-type.
-static std::unique_ptr <STBase> parseLeaf (
+static boost::optional<detail::STVar> parseLeaf (
     std::string const& json_name,
     std::string const& fieldName,
     SField const* name,
     Json::Value const& value,
     Json::Value& error)
 {
-    std::unique_ptr <STBase> ret;
+    boost::optional <detail::STVar> ret (boost::none);
```

Comment:
Info: _Pretty sure_ the explicit construction isn't necessary here. Doesn't hurt, and probably makes the code a bit more readable.